docs: update to account for CLI changes

firrae opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?
This PR updates the documentation to account for the removal of the `--module` flag for the CLI's generate service command.

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #25196


## What is the new behavior?


## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information
Updated the language around the tutorial for services to specify that providedIn: Module should only be used in code libraries and not your general application.

Updated the language used in the dependency injection guide to clarify that 'root' should be the default value for providedIn for most use cases, but that providedIn: Module is available for rare cases of isolation like libraries.

I am openly looking for feedback on this change to figure out the best wording.

firrae commented:
I signed it!

brandonroberts commented:
Part of me thinks the command sample should just be removed altogether and just point to the dependency injection guide to learn more about providers since we already have a rewrite in progress that covers it in detail.

firrae commented:
@brandonroberts I agree, I just wasn't sure how far to go with changing the structure. I will remove it all together. Would you like me to undo the changes to the dependency injection page as well if there is a proper rewrite coming for it?

mary-poppins commented:
You can preview 3f50961 at https://pr25223-3f50961.ngbuilds.io/.
You can preview c596527 at https://pr25223-c596527.ngbuilds.io/.
You can preview 52ef612 at https://pr25223-52ef612.ngbuilds.io/.

brandonroberts commented:
Yes. Just update the note to say "To learn more about providers ...".

brandonroberts commented:
And remove `(aio)` from the git commit message so its just `docs: ...`

mary-poppins commented:
You can preview 6e60392 at https://pr25223-6e60392.ngbuilds.io/.

mary-poppins commented:
You can preview 7c4f031 at https://pr25223-7c4f031.ngbuilds.io/.

jenniferfell commented:
Hi. Thanks for the collaboration on this. I agree with both decisions here: 
- Remove changes to DI guide, because a larger PR is in progress
- Remove the details from the tutorial note. It's too much info here and not immediately relevant to the example.

My only thought is this: Should we also direct readers to the Providers section of ngModules: 

To learn more about providers, see the Providers section (https://angular.io/guide/providers). To learn more about injectors, see the Dependency Injection guide.

jbogarthyde commented:
The new DI guide (which I hope is almost ready to go) has a whole page on the injector hierarchy, and another whole page on the ins and outs of how and where to specify providers. 
I would love your feedback on whether the points you want to make are covered.

Check out PR #21915

firrae commented:
@jenniferfell I think there is a benefit currently to doing so and I can make that change. Though it seems to me that Angular is moving towards using `providedIn` over `providers: []` so I wonder if maybe it encourages developers in a direction counter to what is hoped for forward progression? I'm on the outside looking in though, so maybe the push to moving over isn't as big as it seems to me at this point.

Like I said though, if you want it added, it will be done.

firrae commented:
@jenniferfell I will go in and add the reference to providers as you mentioned. I had originally thought, once again through inference from discussions in issues, that `providers` wasn't long for this world so I initially thought it would be best to push people to DI instead. Thanks to @jbogarthyde for referencing those preview docs as that has begun to clear things up for me as I continue to read the new content.

mary-poppins commented:
You can preview 9c67816 at https://pr25223-9c67816.ngbuilds.io/.

brandonroberts commented:
@firrae changes LGTM. Will you squash down to a single commit wit the original commit message?

firrae commented:
@brandonroberts I will do that this morning for you.

firrae committed:
docs: update to account for CLI changes

This should help clarify the use of providedIn and correct the documentation where it was showing the use of a now depreciated CLI command flag.

I am openly looking for feedback on this change to figure out the best wording.

firrae commented:
OK, it took me 2 tries but I think I squashed them together into a single one now.

mary-poppins commented:
You can preview 40c566b at https://pr25223-40c566b.ngbuilds.io/.

mary-poppins commented:
You can preview 2b125e6 at https://pr25223-2b125e6.ngbuilds.io/.
